Change PersonalizationTags type

kendaleiv opened the pull request:
Change PersonalizationTags type from `IDictionary<string, string>` to `IEnumerable<TransactMessageRecipientPersonalizationTag>`.

kendaleiv committed:
Change PersonalizationTags type to IEnumerable<...>

kendaleiv committed:
Change 3M performance test to 5M

jrusbatch commented on test/Silverpop.Core.Performance/Program.cs:
```diff
@@ -38,9 +38,9 @@ private static void Main(string[] args)
+            Console.WriteLine("Testing 5 million recipients with {0} tags using batches of 5000:", numberOfTags);
             new TransactMessagePerformance()
-                .InvokeGetRecipientBatchedMessages(3000000, 5000, personalizationTags);
+                .InvokeGetRecipientBatchedMessages(5000000, 5000, personalizationTags);
```
It would be better to pull these numbers out into constants so you only need to change them in one place in the future.

jrusbatch replied:
I recommend pulling the number of recipients out into a constant, and using a multiplier here.

kendaleiv committed:
Move hardcoded values into static fields

- Slightly changes the output, no longer prints
  "1 million" and instead prints "1000000", etc.

jrusbatch commented:
:sparkles:
